test-case/check-signal-stop-start.sh

```diff
@@ -95,6 +95,8 @@ do
     dlogi "Testing: run stop/start test on PCM:$pcm,$pipeline_type. Interval time: $interval"
     dlogc $cmd -D$dev -r $rate -c $channel -f $fmt $dummy_file -q &
     $cmd -D$dev -r $rate -c $channel -f $fmt $dummy_file -q & pid=$!
+    [[ $? -ne 0 ]] && dloge "$cmd error without process" && exit 1
```

Comment:
the `$?` mapping to the last command result, so it will be `pid=$!`  
I don't think this check will work as your expect

you can try this command for $? usage
```
[[ "$(ls)" ]] && echo "here"
echo $?
```

Reply:
@Bin-QA Thanks the info re: `$(ls)` --- I didn't know of a way to handle this since the aplay / arecord command ends in ` & pid=$!`

Reply:
@Bin-QA This was good comment. I was testing error cases in locally, it took sometime.
Catching return value for background process is not good idea. Rather we should check for the process id still running after short sleep. I will submit a patch shortly.

Reply:
@Bin-QA  @fredoh9 will it better to assign the cmd execution result to some variable, and make a check?

Reply:
`$?` usage example:
```
[[ "string" ]] && echo "here"
echo $?
[[ "" ]] && echo "here"
echo $?
```

So here `$?` is use for `exit status`  
if you want to know background process exit status ,you can try `wait` command